agent/agent.go

```diff
@@ -18,6 +18,12 @@ const (
 	nodeUpdatePeriod             = 20 * time.Second
 )
 
+// Secrets is a map that keeps all the currenty available secrets to the agent
+type Secrets struct {
+	sync.RWMutex
```

Comment:
It's a good idea to unembed the mutex as a private field so callers in other packages can't directly lock or unlock it.

```go
mu sync.RWMutex
```

Or even better, unexport the `Secrets` type if it will only be used in this package.